new-frontiers-14/frontier-station-14
# Add an auto-labeller to the chem dispenser

## About the PR



Adds a togglable auto-labeller to chemical dispensers.  When on, this labels any jug inserted into the chem dispenser with its primary reagent, and, if impure, its relative purity (see Media below).

## Why / Balance


Manually labelling jugs of chemicals is annoying, this gives chemists some quality of life.  Considering that this information (edit: and more!) is known if you have a ChemMaster or chemistry glasses, this doesn't seem overpowered.

## How to test


1. Spawn a chemical dispenser.  Examine it, it should say the auto-labeler is on.
2. Spawn a plain jug, add your beverage of choice.  Insert the jug into the chemical dispenser.
3. Open the chemical dispenser UI.  The beverage jug should be properly labelled (e.g. "root beer").
4. Right-click, select "Turn off auto-labeler".
5. Repeat step 2.
6. Open the chemical dispenser UI.  The new beverage jug should just be labelled "jug".
7. Examine the chemical dispenser, the panel should say the auto-labeler is off.

## Media

A picture of the chemical dispenser in use.  The "mead (53.84%)" and "iced tea" jugs were auto-labelled when inserted.  The examine panel near the bottom of the screenshot shows that the auto-labeller is on.
![image](https://github.com/user-attachments/assets/221078c6-f952-4bbd-b45d-bd6a0c4e5886)

## Requirements



## Breaking changes


**Changelog**


:cl:
- add: The chemical dispenser has a togglable auto-labeler that labels inserted jugs.
